src/main.tsx

```diff
@@ -1,20 +1,54 @@
 import './index.css';
 
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { invoke } from '@tauri-apps/api';
 import { Provider } from 'jotai';
-import React from 'react';
+import React, { useState } from 'react';
 import ReactDOM from 'react-dom/client';
 
 import App from './application/App';
+import { useAsyncEffect } from './hooks/useAsyncEffect';
+import { ensureProjectFileStructure } from './io/filesystem';
+import { noop } from './lib/noop';
+import { notifyInfo } from './notifications/notifications';
+import { interceptConsoleMessages } from './notifications/notifications.console';
+import { initSettings } from './settings/settingsManager';
 
 const queryClient = new QueryClient();
 
 ReactDOM.createRoot(document.getElementById('root')!).render(
   <React.StrictMode>
     <Provider>
       <QueryClientProvider client={queryClient}>
-        <App />
+        <AppStartup />
       </QueryClientProvider>
     </Provider>
   </React.StrictMode>,
 );
+
+function AppStartup() {
+  const [initialized, setInitialized] = useState(false);
+
+  useAsyncEffect(
+    async () => {
+      // Note: Intercepting INFO, WARN and ERROR console.*
+      interceptConsoleMessages(true, true, true);
+
+      notifyInfo('Application Startup');
+      await initSettings();
+      await ensureProjectFileStructure();
+      await invoke('close_splashscreen');
+      notifyInfo('Application Initialized');
+
+      setInitialized(true);
+    },
```

Comment:
I think you should use `isMounted`:
```typescript
async (isMounted) => {
      // Note: Intercepting INFO, WARN and ERROR console.*
      interceptConsoleMessages(true, true, true);

      notifyInfo('Application Startup');
      await initSettings();
      await ensureProjectFileStructure();
      await invoke('close_splashscreen');
      notifyInfo('Application Initialized');

      if (isMounted()) {
        setInitialized(true);
      }
    },
    noop,
    [],
  );
```
Even though it shouldn't be possible to unmount `AppStartup`, I think this is safer.